source/server/server.cc

```diff
@@ -248,6 +248,9 @@ void InstanceImpl::initialize(Options& options,
   listener_manager_.reset(
       new ListenerManagerImpl(*this, listener_component_factory_, worker_factory_));
 
+  // Shared storage of secrets from SDS
+  secret_manager_.reset(new Secret::SecretManagerImpl());
```

Comment:
You should be able to initialize this in the initializer list.

Reply:
Moved to the initializer list